examples/spawn-at-surface-with-anchors.html

```diff
@@ -0,0 +1,277 @@
+  // until the first hit is found, and move it there
+  var geometry = new THREE.BoxGeometry(BOX_SIZE, BOX_SIZE, BOX_SIZE);
+  var faceIndices = ['a', 'b', 'c'];
+  for (var i = 0; i < geometry.faces.length; i++) {
+    var f  = geometry.faces[i];
+    for (var j = 0; j < 3; j++) {
+      var vertexIndex = f[faceIndices[ j ]];
+      f.vertexColors[j] = colors[vertexIndex];
+    }
+  }
+  // Shift the cube geometry vertices upwards, so that the "pivot" of
+  // the cube is at it's base. When the cube is added to the scene,
+  // this will help make it appear to be sitting on top of the real-
+  // world surface.
+  // geometry.applyMatrix( new THREE.Matrix4().setTranslation( 0, BOX_SIZE / 2, 0 ) );
+  geometry.translate( 0, BOX_SIZE / 2, 0 );
+  var material = new THREE.MeshBasicMaterial({ vertexColors: THREE.VertexColors });
+  cube = new THREE.Mesh(geometry, material);
+
+  // Initialize the anchor manager
+  anchorManager = new THREE.ARAnchorManager(vrDisplay);
+  // Just show a message when anchors update.
+  anchorManager.addEventListener("anchorsupdated", function(event) {
+    console.log(event.anchors.length + " Object3D-s updated their pose!");
+  });
+
+  // Bind our event handlers
+  window.addEventListener('resize', onWindowResize, false);
+  window.addEventListener('touchstart', onClick, false);
+
+  // Kick off the render loop!
+  update();
+}
+
+/**
+ * The render loop, called once per frame. Handles updating
+ * our scene and rendering.
+ */
+function update() {
+  // Clears color from the frame before rendering the camera (arView) or scene.
+  renderer.clearColor();
+
+  // Render the device's camera stream on screen first of all.
+  // It allows to get the right pose synchronized with the right frame.
+  arView.render();
+
+  // Update our camera projection matrix in the event that
+  // the near or far planes have updated
+  camera.updateProjectionMatrix();
+
+  // Update our perspective camera's positioning
+  vrControls.update();
+
+  // Render our three.js virtual scene
+  renderer.clearDepth();
+  renderer.render(scene, camera);
+
+  // Kick off the requestAnimationFrame to call this function
+  // when a new VRDisplay frame is rendered
+  vrDisplay.requestAnimationFrame(update);
+}
+
+/**
+ * On window resize, update the perspective camera's aspect ratio,
+ * and call `updateProjectionMatrix` so that we can get the latest
+ * projection matrix provided from the device
+ */
+function onWindowResize () {
+  camera.aspect = window.innerWidth / window.innerHeight;
+  camera.updateProjectionMatrix();
+  renderer.setSize(window.innerWidth, window.innerHeight);
+}
+
+/**
+ * When clicking on the screen, fire a ray from where the user clicked
+ * on the screen and if a hit is found, place a cube there.
+ */
+function onClick (e) {
+  // Remove cubes with more than one finger.
+  if (e.touches.length > 1 && cubes.length > 0) {
+    scene.remove(cubes[0]);
+    anchorManager.remove(cubes[0]);
+    cubes.splice(0, 1);
```

Comment:
If we're popping off the first element of the array, maybe `cubes.shift()` might be more intuitive